sycl/test-e2e/KernelCompiler/kernel_compiler_sycl.cpp

```diff
@@ -0,0 +1,167 @@
+    // Handler.set_arg(1, start);
+    // Handler.set_arg(2, Range);
+    Handler.parallel_for(R1, Kernel);
+  });
+  Queue.wait();
+
+  for (int i = 0; i < Range; i++) {
+    std::cout << usmPtr[i] << " ";
+    // assert(usmPtr[i] = i + 42);
+  }
+  std::cout << std::endl;
+
+  sycl::free(usmPtr, Queue);
+}
+
+void test_build_and_run() {
+  namespace syclex = sycl::ext::oneapi::experimental;
+  using source_kb = sycl::kernel_bundle<sycl::bundle_state::ext_oneapi_source>;
+  using exe_kb = sycl::kernel_bundle<sycl::bundle_state::executable>;
+
+  sycl::queue q;
+  sycl::context ctx = q.get_context();
+
+  bool ok =
+      q.get_device().ext_oneapi_can_compile(syclex::source_language::sycl);
+  if (!ok) {
+    std::cout << "Apparently this device does not support SYCL source "
+                 "kernel bundle extension: "
+              << q.get_device().get_info<sycl::info::device::name>()
+              << std::endl;
+    return;
+  }
+
+  // create from source
+  source_kb kbSrc = syclex::create_kernel_bundle_from_source(
+      ctx, syclex::source_language::sycl, SYCLSource,
+      syclex::properties{syclex::include_files{"AddEm.h", AddEmH}});
+
+  // double check kernel_bundle.get_source() / get_backend()
+  sycl::context ctxRes = kbSrc.get_context();
+  assert(ctxRes == ctx);
+  sycl::backend beRes = kbSrc.get_backend();
+  assert(beRes == ctx.get_backend());
+
+  // compilation of empty prop list, no devices
+  exe_kb kbExe1 = syclex::build(kbSrc);
+
+  // compilation with props and devices
+  std::string log;
+  std::vector<std::string> flags{"-g", "-fno-fast-math"};
+  std::vector<sycl::device> devs = kbSrc.get_devices();
+  exe_kb kbExe2 = syclex::build(
+      kbSrc, devs,
+      syclex::properties{syclex::build_options{flags}, syclex::save_log{&log}});
+  assert(log.find("warning: 'this_nd_item<1>' is deprecated") !=
+         std::string::npos);
+
+  sycl::kernel k = kbExe2.ext_oneapi_get_kernel("__sycl_kernel_ff_cp");
+
+  // COMING SOON
+  // sycl::kernel_bundle<sycl::bundle_state::executable> kb
+  //    = syclexp::build(kb_src,
+  // syclexp::properties{syclexp::registered_kernel_names{"mykernels::bar"}});
+  // sycl::kernel k = kb.ext_oneapi_get_kernel("mykernels::bar");
+
+  // 4
+  test_1(q, k);
+}
+
+void test_error() {
+  namespace syclex = sycl::ext::oneapi::experimental;
+  using source_kb = sycl::kernel_bundle<sycl::bundle_state::ext_oneapi_source>;
+  using exe_kb = sycl::kernel_bundle<sycl::bundle_state::executable>;
+
+  sycl::queue q;
+  sycl::context ctx = q.get_context();
+
+  bool ok =
+      q.get_device().ext_oneapi_can_compile(syclex::source_language::sycl);
+  if (!ok) {
+    return;
+  }
+
+  source_kb kbSrc = syclex::create_kernel_bundle_from_source(
+      ctx, syclex::source_language::sycl, BadSource);
+  try {
+    exe_kb kbExe = syclex::build(kbSrc);
+    assert(false && "we should not be here");
+  } catch (sycl::exception &e) {
+    // yas!
+    assert(e.code() == sycl::errc::build);
```

Comment:
Maybe check something about the error string also, similar to how you check for the warning message in the build log?